From fe772763a5c96427249f154f1ce9778ed617ff6e Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Thu, 30 May 2019 09:22:38 -0400
Subject: [PATCH] ENH(TST): adding one more condition (for nd80) to skip
 testing for url to be in error msg

Failed test run looks like

	======================================================================
	FAIL: datalad.interface.tests.test_download_url.test_download_url_exceptions
	----------------------------------------------------------------------
	Traceback (most recent call last):
	  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
		self.test(*self.arg)
	  File "/build/datalad-0.11.5/.pybuild/pythonX.Y_2.7/build/datalad/interface/tests/test_download_url.py", line 47, in test_download_url_exceptions
		assert_in('http://example.com/bogus', msg)
	AssertionError: 'http://example.com/bogus' not found in u"Failed to establish a new session 1 times. Last exception was: HTTPConnectionPool(host='example.com', port=80): Max retries exceeded with url: /bogus (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f5ecc023250>: Failed to establish a new connection: [Errno -2] Name or service not known',)) [adapters.py:send:423] [http.py:get_downloader_session:502]"
	-------------------- >> begin captured logging << --------------------
	requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): example.com
	--------------------- >> end captured logging << ---------------------
---
 datalad/interface/tests/test_download_url.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/datalad/interface/tests/test_download_url.py b/datalad/interface/tests/test_download_url.py
index 23ad0ce58..8440f0b12 100644
--- a/datalad/interface/tests/test_download_url.py
+++ b/datalad/interface/tests/test_download_url.py
@@ -43,6 +43,8 @@ def test_download_url_exceptions():
         'Cannot connect to proxy.' in msg
         or
         'Temporary failure in name resolution' in msg
+        or
+        'Name or service not known' in msg  # nd80
     ):
         assert_in('http://example.com/bogus', msg)
 
-- 
2.21.0.593.g511ec345e18

